### DISCUSSION TITLE
FTP Server for HA?

### DISCUSSION THREAD

**Discussion by guarddog13** [tteck/Proxmox#576] [Ideas]
Is there anyway to have the FTP addon as an lxc?Would this even work? I have moved all my addons off of Home Assistant OS as I want to try one of the HA LXCs. The FTP addon is one of two I haven't been able to figure out a solution for without the supervisor.

https://github.com/hassio-addons/addon-ftp/blob/v4.6.1/README.md

**Comment by guarddog13** [comment]
Could it be setup like the Filebrowser is? Where the script is run in the LXC and installs FTP?

**Reply by tteck** [reply]
Maybe a LXC with [SFTPgo](https://github.com/drakkan/sftpgo)?

**Reply by guarddog13** [reply]
Can you put that in a script? 😄 Whatever works so that I can still send my video clips to the folder I have setup on HA as a folder_watcher. 

On an unrelated noted....I can't get ambientweather2mqtt (i'm currently using an addon option called awnet to local) to work or I would request that as an LXC script as well. lol.

**Reply by guarddog13** [reply]
> Maybe a LXC with [SFTPgo](https://github.com/drakkan/sftpgo)?

So I tried this. I also tried a couple of other FTPs. I tried them in my test HA Container LXC. So what happens is the two docker containers are separate from each other. HA can't see where the FTP files go and the FTP server can't put files into the HA Container. If for example I put the path as the media folder in HA it just creates that path in the FTP volume. So of course HA can't do a folder_watcher of this folder. 

Somehow the FTP has to be inside the HA file system. I wonder...is there an FTP that is run as a system? Seems it would have better access to the docker volumes...

**Reply by andreapier** [reply]
@tteck Would it be possible to create a script featuring alpine + vsftp (maybe with virtual users enabled if it's not too much)?  
That is quite a challenge, at least for me. It would make a nice low resource LXC container but it would work just fine
